Jenkinsfile

```diff
@@ -25,19 +15,99 @@ pipeline {
     }    
 
     stages {
-        stage('Run SonarQube analysis') {
+        stage('Build and Test Locally') {
+            stages {
+                stage('Build Docker Images') {
+                    steps {
+                        script {
+                            sh label: 'Stopping running containers (preventative maintenance)', script: 'docker-compose down -v'
+                            sh label: 'Running docker-compose build', script: 'docker-compose build --build-arg BUILD_TAG=${BUILD_TAG}'
+                        }
+                    }
+                }
+                stage('Deploy Locally') {
+                    steps {
+                        sh label: 'Starting containers', script: 'docker-compose up -d rabbitmq dynamodb nhais'
+                        echo "Waiting 10 seconds for containers to start"
+                        sleep 10
+                        sh label: 'Show all running containers', script: 'docker ps'
+                    }
+                }
+                stage('Run tests') {
+                    steps {
+                        sh label: 'Running tests', script: 'docker-compose run nhais-tests'
+                    }
+                    post {
+                        always {
```

Comment:
What is the reason for `post { always {` here? could it be part of the main stage steps?

Reply:
To ensure test reports were copied, when a test fails the rest of the build fails

Reply:
ok, understood